elastic/kibana
# [Stateful sidenav] Add deeplink definitions & hide page side nav

In this PR I've added the missing Deeplink declarations required for the new side navigation definitions (search, oblt, security).

I've also updated the logic in the different apps (management, ml, security) that use the "solutionNav" prop of the `KibanaPageTemplate` to include their nav.

We can rely on the core chrome style ("classic" or "project") to determine if the side navigation should be visible or not. When not in classic, we don't want to show it.

## How to test

* Start Kibana as usual and make sure that the side navigation of your app is still there
* Then add the following in the `kibana.yml` file

```yml
navigation.solutionNavigation.featureOn: true
navigation.solutionNavigation.enabled: true
navigation.solutionNavigation.optInStatus: visible
navigation.solutionNavigation.defaultSolution: es
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"
```

* Refresh the browser, you will be in the new navigation.

As the nav tree will come in a separate PR use this links to check that the sideNav of your app does not render anymore

- ml: `app/ml/jobs`
- management: `app/management/data/index_management/indices`
- security: `app/security/rules/landing`
